### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Timowest12**
In this pull request:

- Added tests for the calculator and operator files
- Added snapshot tests for the components

### CODE REVIEW

File: src/__tests__/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  test('test divide', () => {
+    expect(operate(10, 5, '÷')).toBe('2');
+  });
+  test('test multiply', () => {
+    expect(operate(10, 5, 'x')).toBe('50');
+  });
+  test('test modules', () => {
+    expect(operate(10, 3, '%')).toBe('1');
+  });
+});
```
Comment (Emmyn5600) [inline src/__tests__/operate.test.js]: - [x] Good job here, however, let us add more tests to move forward.

- Should throw an error if given a wrong number
- Should give a 0 error if given a wrong operator

File: src/__tests__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    expect(endres.total).toBe('30');
+  });
+  test('test modules', () => {
+    let endres = calculate(obj, '10');
+    endres = calculate(endres, '%');
+    endres = calculate(endres, '3');
+    endres = calculate(endres, '=');
+    expect(endres.total).toBe('1');
+  });
+});
```
Comment (Emmyn5600) [inline src/__tests__/calculate.test.js]: - [x] Implement the following tests.

- should return 0 as next if the buttonName is AC
- should throw 0 error if trying to make a zero division
- should throw Zero division error

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Timowest12 and @mwanawabangona 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.


 ## Required Changes ♻️
 
_Check the comments under the review._

## Optional suggestions

- N/A 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/__tests__/calculate.test.js (hunk 2 above)
Comment (Timowest12) [reply src/__tests__/calculate.test.js]: Added test for ac button and zero devision

File: src/__tests__/operate.test.js (hunk 1 above)
Comment (Timowest12) [reply src/__tests__/operate.test.js]: Added test that checks for correct numbers and operators

**Review by o-rashad — APPROVED** [review]
Asslam Alaikum Team @Timowest12 @mwanawabangona  ,😊

Hi👋🏽, Thank you for implementing the changes required by the previous reviewer :100: :ok_hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it 😎 

# Congratulations! 🎉 

Happy coding!👏

#### To mention :exclamation: 
 > Please make sure to use pair programing next time :handshake: [LINK](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md#:~:text=you%20need%20to%20use%20pair-programming%20for%20this%20project.%20we%20will%20check%20to%20see%20if%20both%20partners%20submitted%20code%20by%20looking%20at%20the%20commit%20history%20for%20each%20project.%20therefore%20you%20need%20to%20use%20only%20one%20repository%20for%20your%20pair-programming%20group.) 
 > index


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**